gremlin-go/driver/gorillaTransporter.go

```diff
@@ -20,42 +20,50 @@ under the License.
 package gremlingo
 
 import (
+	"crypto/tls"
 	"net/url"
-	"strconv"
+	"time"
 
 	"github.com/gorilla/websocket"
 )
 
+const maxFailCount = 3
+
 // Transport layer that uses gorilla/websocket: https://github.com/gorilla/websocket
 // Gorilla WebSocket is a widely used and stable Go implementation of the WebSocket protocol.
 type gorillaTransporter struct {
-	host       string
-	port       int
+	url        string
 	connection websocketConn
 	isClosed   bool
+	authInfo   *AuthInfo
+	tlsConfig  *tls.Config
 }
 
+// Connect used to establish a connection.
 func (transporter *gorillaTransporter) Connect() (err error) {
 	if transporter.connection != nil {
 		return
 	}
 
-	u := url.URL{
-		Scheme: scheme,
-		Host:   transporter.host + ":" + strconv.Itoa(transporter.port),
-		Path:   path,
+	var u *url.URL
+	u, err = url.Parse(transporter.url)
+	if err != nil {
+		return
 	}
 
 	dialer := websocket.DefaultDialer
+	dialer.TLSClientConfig = transporter.tlsConfig
 	// TODO: make this configurable from client; this currently does nothing since 4096 is the default
 	dialer.WriteBufferSize = 4096
-	conn, _, err := dialer.Dial(u.String(), nil)
+	// Nil is accepted as a valid header, so it can always be passed directly through.
+	conn, _, err := dialer.Dial(u.String(), transporter.authInfo.getHeader())
```

Comment:
is there a timeout here for cases when the server is alive but the connection is taking longer than usual?

Reply:
You can set a timeout in gorilla but currently we are not using it, we found the timeout structure of gorilla to be problematic. It will require a bit more effort and will likely be done in a future milestone.

The reason it is problematic is because if you use it and it expires, the connection becomes unusable - which is okay when we have pooling and recovery set up, but not in our current simple single connection model.